docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md

```diff
@@ -80,13 +83,346 @@ This example demonstrates how to configure the order state machine transition fr
+Install or upgrade the modules to the specified or later versions:
+- `spryker/kernel-app:1.2.0`
+- `spryker/payment:5.24.0`
+- `spryker/payments-rest-api:1.3.0`
+
+#### PreOrder payment flow
+
+1. The customer either selects Stripe as the payment method or Stripe Elements is loaded by default.
+2. When Stripe is selected, the `InitializePreOrderPayment` Glue API endpoint (`glue.mysprykershop.com/payments`) is called with the following parameters:
+  * Payment provider name: Stripe
+  * Payment method name: Stripe
+  * Payment amount
+  * Quote data
+3. Back Office sends the quote data and the payment amount to Stripe app through an API.
+4. The payment with the given data is persisted in the Stripe app.
+5. Stripe app requests `ClientSecret` and `PublishableKey` keys from Stripe through an API.
+6. Stripe returns a JSON response with the following parameters:
+  * TransactionId
+  * ClientSecret
+  * PublishableKey
+  * Only for marketplaces: AccountId
+7. Stripe Elements is rendered on the order summary page. See [Example of rendering Stripe Elements on the summary page](#example-of-rendering-stripe-elements-on-the-summary-page) for rendering Stripe Elements.
+8. The customer selects a payment method in Stripe Elements and submits the data.
+9. The customer is redirected to the configured `return_url`, which makes an API request to persist the order in the Back Office: `glue.mysprykershop.com/checkout`.
+10. The customer is redirected to the application's success page.
+11. Stripe app confirms the PreOrder payment using the plugin: `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin`.
+  The `order_reference` is passed to the Stripe app to be connected with `transaction_id`.
+12. Stripe app processes the payment and sends a `PaymentUpdated` message to Spryker.
+  The message contains additional data, which you can see in the Back Office.
+13. Depending on payment status, one of the following messages is returned through an asynchronous API:
+  *  Payment is successful: `PaymentAuthorized` message.
+  *  Payment is failed: `PaymentAuthorizationFailed` message.
+14. Further on, the order is processed through the OMS.
+
+All payment related messages mentioned above are handled by `\Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin`, which is registered in `MessageBrokerDependencyProvider`.
+
+### Example integration
+
+To check out how the integration works, see this [example application](https://github.com/spryker-projects/spa-checkout-glue-with-stripe).
+
+Before the customer is redirected to the summary page, all required data is collected: customer data, addresses, and selected shipment method. When the customer goes to the summary page, to get the data required for rendering the Stripe Elements, the application needs to call the `InitializePreOrderPayment` Glue API endpoint.
+
+#### Example of rendering Stripe Elements on the summary page
+
+```JS
```

Comment:
I assume this code is the example of rendering stripe elements on the summary page?

Reply:
No this is the initial request to get the required data from Stripe to render the payment elements. This includes the clientSecret, the publishableKey, and the accountId.